### PR TITLE
Hide code generated from hidden blocks when using "show code"

### PR DESCRIPTION
**Pull request by balderdash**

### CODE REVIEW

File: blockly-core/core/code_generation/generator.js [hunk 1]
```diff
@@ -164,18 +168,21 @@ Blockly.CodeGenerator = function(name) {
...
  *     operator order value.  Returns '' if block is null.
  */
-Blockly.CodeGenerator.prototype.blockToCode = function(block) {
+Blockly.CodeGenerator.prototype.blockToCode = function(block, opt_showHidden) {
   if (!block) {
     return '';
   }
-  if (block.disabled) {
-    // Skip past this block if it is disabled.
+  var showHidden = (typeof opt_showHidden == "undefined") ? true : opt_showHidden;
```
Comment (Hamms) [inline blockly-core/core/code_generation/generator.js]: `opt_showHidden === undefined` should also work, and is a bit simpler.

**Comment by Hamms** [comment]
What did we conclude / how does this function regarding functional definitions?

**Comment by Hamms** [comment]
Great tests and comments! Could you also add a test verifying that `opt_showHidden` defaults to true? Otherwise LGTM!

**Comment by balderdash** [comment]
Function definitions will show up as long as they don't have the uservisible=false attribute. I found and fixed 30 levels with needlessly hidden function definitions on Friday, so the JS version should always be visible as long as the block version is visible (whether in the main blockspace or in the function editor).

File: blockly-core/core/code_generation/generator.js (hunk 1 above)
Comment (balderdash) [inline blockly-core/core/code_generation/generator.js]: Makes sense, done.

**Pull request merged**